sdk/data/azcosmos/cosmos_container.go

```diff
@@ -41,6 +41,9 @@ func (c *ContainerClient) ID() string {
 func (c *ContainerClient) Read(
 	ctx context.Context,
 	o *ReadContainerOptions) (ContainerResponse, error) {
+	var err error
+	ctx, endSpan := runtime.StartSpan(ctx, "ContainerClient.Read", c.database.client.internal.Tracer(), nil)
```

Comment:
Though this is consistent with the other SDKs, I am curious if we should be more explicit here with the span name?
Also, in other Cosmos SDKs, we add additional information on the span names, like container id in this case.

From Java SDK - `this.readContainerSpanName = "readContainer." + this.id;`

I wonder if this has been considered in GO language? It is based on the feedback from customers, it really helps them to see which database / container these operations are for?

Reply:
In Go, it seems that's the pattern they've used:

https://github.com/Azure/azure-sdk-for-go/blob/f53cbcb8e060bc481a0893e003fbcf1dfddbf28e/sdk/data/aztables/service_client.go#L118

@jhendrixMSFT do you have any context?

Reply:
Thanks @ealsur, tagging @lmolkova to provide her guidance here.

Reply:
CosmosDB span naming pattern is described here https://github.com/open-telemetry/semantic-conventions/blob/v1.27.0/docs/database/database-spans.md. The format has slightly changed recently and it's slightly different from what Java and .NET populate.

We can stay consistent with Cosmos DB .NET/Java for now and update them all together later on or implement the latest version of OTel semantic conventions in Go and tackle Java and .NET later on. 

According to the latest version, the span name should be `readContainer {container name}` when performing operations within the container and `{operation} {database}` when doing some operations against the whole database.

The otel semconv take a priority over Azure SDK naming conventions, but otel semconv are defined for a (relatively) small subset of SDKs.

Reply:
Added span names following https://github.com/ealsur/azure-sdk-for-go/blob/cd4154850235edd9165a4b72f5e21cbeacc97ef6/sdk/data/azcosmos/otel_constants.go#L9-L29